frontend/packages/dev-console/src/components/pipelines/modals/common/PipelineResourceDropdown.tsx

```diff
@@ -41,21 +42,27 @@ const PipelineResourceDropdown: React.FC<PipelineResourceDropdownProps> = (props
     resourceDefinition,
   );
 
-  const availableResources: PipelineResourceKind[] = resources.filter(
-    (resource) => resource.spec.type === filterType,
+  const availableResources: PipelineResourceKind[] = React.useMemo(
+    () => resources.filter((resource) => resource.spec.type === filterType),
+    [filterType, resources],
   );
 
-  const canAutoSelect = autoSelect && !touched && loaded && !error;
+  const canAutoSelect = autoSelect && !touched && loaded && !error && !!pipelineName;
   React.useEffect(() => {
     if (canAutoSelect) {
-      if (availableResources.length === 0) {
-        setFieldValue(name, CREATE_PIPELINE_RESOURCE);
+      const matchingResources = availableResources.filter(
+        (resource) => resource.metadata?.labels?.['tekton.dev/pipeline'] === pipelineName,
+      );
+      if (matchingResources.length > 0) {
+        setFieldValue(name, matchingResources[0].metadata.name, false);
+      } else if (availableResources.length > 0) {
+        setFieldValue(name, availableResources[0].metadata.name, false);
```

Comment:
@andrewballantyne I keep the "select the first resource" fallback here. If we drop this here the user needs to select a resource if there is no matching resource found. I would prefer this over selecting "a random one". Because of this question and that we are close to the release I marked this PR as WIP.